### PR TITLE
Sweet Sundae Ramen

### PR DESCRIPTION
**Pull request by Kurfursten**
- Adds a new recipe to cook ramen cups (5u Flour + Cardboard)
- Adds a new recipe to cook sundae ramen (30u dry_ramen, 1u black food coloring, 1u sprinkles, 6u hardcores, 1 banana, 1 chocolate bar)
- Adds a new recipe to cook sweet sundae ramen (1 sundae ramen, 1 giga puddi, 1 donk pocket, 1 rice pudding)
- Adds a new food, sundae ramen: contains 10u nutriment and 2u dan sauce
- Adds a new food, sweet sundae ramen: contains 70u procedurally generated reagents (which changes its description). Reagents are selected in 10u increments with these chances:
  - 50% nutriment
  - 10% tricordrazine
  - 10% oxycodone
  - 10% mint toxin
  - 10% mindbreaker
  - 10% zombie powder

---
- Special thanks to: anonymous spriter, anonymous /ck/ poster
- Fully tested

### CODE REVIEW

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -3369,3 +3369,56 @@
...
+			reagents.add_reagent("zombiepowder", 10)
+		if(2)
+			desc += " This may not be everyone's cup of tea, but it's great, I promise."
+			reagents.add_reagent("oxycodone", 10)
+		if(3)
+			desc += " This has the cook's favorite ingredient -- and a lot of it!"
+			reagents.add_reagent("mindbreaker", 10)
+		if(4)
+			desc += " It has TONS of flavor!"
+			reagents.add_reagent("minttoxin", 10)
```
Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: minttoxin?

**Comment by 9600bauds** [comment]
We seriously need a better way to make flour-based recipes so you don't have to use cardboard to make ramen or an egg to make a donut

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (CptWad) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Its the stuff inside mints. If you are fat you gib.

**Comment by FudgePucker** [comment]
Why not be able to make bowls of ramen or blocks of dry ramen?
I want to be able to make super spicy ramen with the recipe being, 30 u ramen, 2 peppers, an egg, and 5 u pepper.
Cardboard isn't the cheapest thing in the game to get a hold of.

Also onions when? I want to make french onion soup with 2 onions toast and cheese. Also able to make onion rings in the food processor that I can deepfry.

**Comment by Kurfursten** [comment]
@FudgePucker Cardboard comes out of the RSF, right? You can just borrow the bartender's -- you even have access to his windoor.

>New ramen types
Cool idea, but not in the scope of the PR, I don't have icons for them.

>Onions
Definitely not in the scope of the PR

**Comment by FudgePucker** [comment]
Yea I know I was just tossing ideas around.
Does cardboard come out of the RSF? I don't remember.

**Comment by Exxion** [comment]
Just flatten your internals box

**Comment by 9600bauds** [comment]
But that internals box is for storing stuff

**Comment by Exxion** [comment]
Flatten one of them, then
You get two
One in the backpack, one in the plastic bag
And you can't even put it back in the bag if you take it out because it's too big

**Comment by Kurfursten** [comment]
I would like to see more cardboard items be flattenable. Wouldn't it be wonderful if you could just flatten out a ramen cup, paper tray, or cardborg head to use for cardboard once more? I might make a separate PR for that right now...

**Comment by FudgePucker** [comment]
Ramen cups aren't even made out of cardboard, cardboard is a horrible choice for holding liquids.
There isn't stacks of styrofoam so I guess the next best thing would be to have the recipe require a sheet of plastic :^)

**Comment by Kurfursten** [comment]
Space Cardboard is wonderful, so much better than our earthly cardboard.

Besides, who microwaves plastic or Styrofoam?

**Comment by FudgePucker** [comment]
What was the recipe that 5 u flour creates?
I am on my phone so I can't check but I think its spaghetti. If it isn't used, then requiring cardboard would be silly.
You don't have to put in a sheet of metal for recepies that come with trays, you don't have to put in wood for recipes that create boals.
The microwaves magical matter generator can handle creating wood and metal, but cardboard is too advanced?

**Comment by Kurfursten** [comment]
Yeah, spaghetti is the 5u flour. This recipe is fine, I think, you've got an RSF next door for all the cardboard you want.

**Comment by FudgePucker** [comment]
Crazy ass idea I just had laying in bed, this is wayyy off of the scale of this PR, and 10 times more complex than goof kitchen.
But what if we had a pasta roller?
Rather than having crazy recipes for things, you would make dough and then put it in the roller then decide what you want.
This makes more sense than 5 flour units making spaghetti from a microwave.
It would also add in backdoor compatibility if someone wants more complex pasta dishes.
Only problem might be complexity and adding yet another fucking machine to the kitchen.

**Comment by Exxion** [comment]
That would probably be frustrating but not really difficult to code.

**Comment by FudgePucker** [comment]
@Exxion
I think with a little overhaul of cooking it could be really interesting.
Dough would have more uses than pasta. You could hand make pizzas with any toppings you want. You could bake bread.

Shit with enough effort we could have a better cooking system than goon.

**Comment by Kurfursten** [comment]
@FudgePucker You're probably better off creating a separate github issue for the feature request

**Comment by Kurfursten** [comment]
@ComicIronic Lacking objections over 2 days, please tag NFD

**Pull request merged**